test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs

```diff
@@ -328,6 +328,53 @@ public void RestoreCommand_FromSolutionFile(string configFileName)
             }
         }
 
+        [Fact]
+        public void RestoreCommand_FromFilteredSolutionFile()
+        {
+            // Arrange
+            var nugetexe = Util.GetNuGetExePath();
+
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var workingPath = pathContext.WorkingDirectory;
+                var repositoryPath = Util.CreateBasicTwoProjectSolutionWithSolutionFilters(workingPath, "packages.config", "packages.config");
+
+                // Act
+                var r = CommandRunner.Run(
+                    nugetexe,
+                    workingPath,
+                    "restore " + Path.Combine(workingPath, "a.proj1.slnf") + " -Source " + repositoryPath,
+                    waitForExit: true);
+
+                // Assert
+                Assert.True(_successCode == r.ExitCode, r.Output + " " + r.Errors);
+                var packageFileA = Path.Combine(pathContext.PackagesV2, "packageA.1.1.0", "packageA.1.1.0.nupkg");
+                var packageFileB = Path.Combine(pathContext.PackagesV2, "packageB.2.2.0", "packageB.2.2.0.nupkg");
+                Assert.True(File.Exists(packageFileA));
+                Assert.False(File.Exists(packageFileB));
+            }
+
+            using (var pathContext = new SimpleTestPathContext())
```

Comment:
Do we need to use a completely separate bath context?

Reply:
The goal is to isolate the two runs because in the first we only want the package A to be restored and in the second the package B. If we only use a unique context the package A will be already restored and the second assert will fail.
The use of the second context was the simplest way that I thought to reset the state of the context.

Reply:
Maybe just make it theory and pass true/false to decide which slnf file to restore.

Reply:
It's ok to have the previous state, as long as we account for that. 

Projects having a different package covers for that. We shouldn't need the separate working directories.